cmd/meroxa/root/apps/apps.go

```diff
@@ -30,6 +31,34 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type ApplicationState string
+
+const (
+	ApplicationStateInitialized ApplicationState = "initialized"
+	ApplicationStateDeploying   ApplicationState = "deploying"
+	ApplicationStatePending     ApplicationState = "pending"
+	ApplicationStateRunning     ApplicationState = "running"
+	ApplicationStateDegraded    ApplicationState = "degraded"
+	ApplicationStateFailed      ApplicationState = "failed"
+
+	// collectionName = "apps".
+)
+
+//var displayDetails = display.Details{"Name": "name", "State": "state", .
+//"SpecVersion": "specVersion", "Created": "created", "Updated": "updated"}.
+
+// Application represents the Meroxa Application type within the Meroxa API.
+type Application struct {
+	ID          string           `json:"id"`
+	Name        string           `json:"name"`
+	State       ApplicationState `json:"state"`
+	Spec        string           `json:"spec"`
+	SpecVersion string           `json:"specVersion"`
```

Comment:
I assume this is part of what we're already discussing in #mdpx, but I think we should continue using snakecase unless there's a very good reason to do so.

```suggestion
	SpecVersion string           `json:"spec_version"`
```